src/main/java/health/ere/ps/service/connector/endpoints/EndpointDiscoveryService.java

```diff
@@ -0,0 +1,201 @@
+    String authSignatureServiceEndpointAddress;
+
+    @ConfigProperty(name = "card-service.endpoint.address", defaultValue = "")
+    String cardServiceEndpointAddress;
+
+    @ConfigProperty(name = "certificate-service.endpoint.address", defaultValue = "")
+    String certificateServiceEndpointAddress;
+
+    @ConfigProperty(name = "event-service.endpoint.address", defaultValue = "")
+    String eventServiceEndpointAddress;
+
+    @ConfigProperty(name = "connector.base-uri")
+    String connectorBaseUri;
+
+    @ConfigProperty(name = "connector.verify-hostname", defaultValue = "true")
+    String connectorVerifyHostname;
+
+    @Inject
+    SecretsManagerService secretsManagerService;
+
+    @PostConstruct
+    void obtainConfiguration() throws IOException, ParserConfigurationException, SecretsManagerException {
+        // code copied from IdpClient.java
+
+        // having an ssl context does not interfere with non-ssl connections
+        SSLContext sslContext = connectorTlsCertTrustStore.isPresent() ? secretsManagerService.createSSLContext(connectorTlsCertTrustStore.get(),
+                connectorTlsCertTrustStorePwd,
+                SecretsManagerService.SslContextType.TLS,
+                SecretsManagerService.KeyStoreType.PKCS12) : secretsManagerService.createAcceptAllSSLContext();
+
+        ClientBuilder clientBuilder = ClientBuilder.newBuilder()
+                .sslContext(sslContext);
+
+        if ("false".equals(connectorVerifyHostname)) {
+            // disable hostname verification
+            clientBuilder = clientBuilder.hostnameVerifier(new SSLUtilities.FakeHostnameVerifier());
+        }
+
+        Invocation invocation = clientBuilder.build()
+                .target(connectorBaseUri)
+                .path("/connector.sds")
+                .request("application/xml")
+                .buildGet();
+
+        try (InputStream inputStream = invocation.invoke(InputStream.class)) {
+            Document document = DocumentBuilderFactory.newDefaultInstance()
+                    .newDocumentBuilder()
+                    .parse(inputStream);
+
+            Node serviceInformationNode = getNodeWithTag(document.getDocumentElement(), "ServiceInformation");
+
+            if (serviceInformationNode == null) {
+                throw new IllegalArgumentException("Could not find single 'ServiceInformation'-tag");
+            }
+
+            NodeList serviceNodeList = serviceInformationNode.getChildNodes();
+
+            for (int i = 0, n = serviceNodeList.getLength(); i < n; ++i) {
+                Node node = serviceNodeList.item(i);
+
+                if (!node.hasAttributes() || node.getAttributes().getNamedItem("Name") == null) {
+                    break;
+                }
+
+                switch (node.getAttributes().getNamedItem("Name").getTextContent()) {
+                    case "AuthSignatureService": {
+                        authSignatureServiceEndpointAddress = getEndpoint(node);
```

Comment:
I don't like override a injected value. Better here: injected value called *Fallback and is selected in getter if value = null

Reply:
This will be changed.